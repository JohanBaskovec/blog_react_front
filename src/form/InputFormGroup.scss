.InputFormGroup {
  display: flex;
  flex-direction: column;
}
