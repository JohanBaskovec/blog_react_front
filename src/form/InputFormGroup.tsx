import React from "react";
import "./InputFormGroup.scss";
import {Label} from "./Label";
import {Input} from "./Input";
import {FieldHelperProps, FieldInputProps, FieldMetaProps, useField} from "formik";
import {ErrorContainer} from "./ErrorContainer";

interface InputFormGroupProps {
    style?: React.CSSProperties;
    children?: React.ReactNode;
    inputName: string;
    inputLabel: string;
}

export function InputFormGroup(props: InputFormGroupProps): JSX.Element {
    const [field, meta]: [FieldInputProps<any>, FieldMetaProps<any>, FieldHelperProps<any>] = useField(props.inputName);
    return (
        <div className="InputFormGroup"
             style={props.style}>
            <Label inputName={props.inputName}>{props.inputLabel}</Label>
            <Input name={props.inputName}/>
            <ErrorContainer style={{margin: "0.5rem 0"}}
                            inputName={props.inputName} />
        </div>
    );
}
