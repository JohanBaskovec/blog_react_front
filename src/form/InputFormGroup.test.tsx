import React from "react";
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'
import {InputFormGroup} from "./InputFormGroup";

describe('InputFormGroup', () => {
    it("is rendered properly", () => {
        const tree = TestRenderer
            .create(<InputFormGroup style={{width: '100%'}}
                                    value='read-only'
                                    inputLabel='input label'
                                    inputName='input-name'>children</InputFormGroup>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
