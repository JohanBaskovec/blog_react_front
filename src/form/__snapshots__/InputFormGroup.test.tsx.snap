// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFormGroup is rendered properly 1`] = `
<div
  className="InputFormGroup"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <label
    className="Label"
    htmlFor="input-name"
  >
    input label
  </label>
  <input
    className="Input"
    id="input-name"
    name="input-name"
    onChange={[Function]}
    value="read-only"
  />
</div>
`;
